src/lib.rs

```diff
@@ -783,6 +804,57 @@ mod test {
             Client::new(),
             server.url().join("andes-1.8.3-pyhd8ed1ab_0.conda").unwrap(),
             check_method,
+            RangeRequestUrlSource::Response,
+            HeaderMap::default(),
+        )
+        .await
+        .expect("bla");
+
+        // Also open a simple file reader
+        let mut file = tokio::fs::File::open(path.join("andes-1.8.3-pyhd8ed1ab_0.conda"))
+            .await
+            .unwrap();
+
+        // Read until the end and make sure that the contents matches
+        let mut range_read = vec![0; 64 * 1024];
+        let mut file_read = vec![0; 64 * 1024];
+        loop {
+            // Read with the async reader
+            let range_read_bytes = range.read(&mut range_read).await.unwrap();
+
+            // Read directly from the file
+            let file_read_bytes = file
+                .read_exact(&mut file_read[0..range_read_bytes])
+                .await
+                .unwrap();
+
+            assert_eq!(range_read_bytes, file_read_bytes);
+            assert_eq!(
+                range_read[0..range_read_bytes],
+                file_read[0..file_read_bytes]
+            );
+
+            if file_read_bytes == 0 && range_read_bytes == 0 {
+                break;
+            }
+        }
+    }
+
+    #[rstest]
+    #[case(RangeRequestUrlSource::Request)]
+    #[case(RangeRequestUrlSource::Response)]
+    #[tokio::test]
+    async fn async_range_reader_url_source(#[case] url_source: RangeRequestUrlSource) {
```

Comment:
I tried to set up a test case that used redirects but it was kind of a pain? I'm not sure what kind of testing you think is valuable here.

Reply:
After I made #15, I also added a commit that has a test for this in my own fork. It adds an endpoint to the unit-test http server that redirects based on the HTTP method used, and that target URL returns `405 Method Not Allowed` if called with another method: https://github.com/torarvid/async_http_range_reader/pull/1

(Didn't want to add it to #15 to keep the PR size manageable, but I can if it's wanted)